### ISSUE TITLE
passing updated man page

### ISSUE THREAD

**Issue by greg-minshall** [kaushalmodi/eless#44]
hi.  eless is very nice.  so, i've been using it a lot.  recently, i was updating a man page for some software of mine which is already installed on my machine.  i would do
````
nroff -man ./credeface.1 2>&1 | eless
````
but, i would always see the *old* man page, in spite of the fact that the new man page was right there.  finally, i realized if i used `less` rather than `eless`, i would see the new man page.

that got me to look at the `eless` source, and it seems you parse the input enough to realize it *is* a man page, then decide *which* man page, then do (the emacs-equivalent of) `man 1 credeface`, so giving me the installed man page, rather than the new one i am working on.

i notice that `M-x man` is willing to take a filename (especially if preceded by a `-l`).

[insert a sentence here on 'i think this is a bug', or 'here is a feature request', or, more likely, 'i don't know what the right thing to do would be'.]

again, thanks for eless!

[edited 02.02.2023, change opinion of `M-x man`.]

**Comment by jibal** [comment]
Yeah, I just discovered the same thing when doing man -l with PAGER=eless

This violates all sorts of software engineering principles, especially on a system based on composable components like UNIX. We expect a pager to display the input that is actually given to it, not to execute a totally different command and display its output. To quote Claude:

"This is completely backwards! Instead of just displaying the content it received, it's trying to be "clever" and re-lookup the man page by name. ... This is definitely a bug in eless - a pager should display what it receives, not try to reverse-engineer and re-fetch it!"

Here's a workaround of sorts: pipe the input through tail -n +1 ... that strips off the first line, defeating eless's logic of figuring out that it's a man page.

**Comment by kaushalmodi** [comment]
@greg-minshall It was tricky to make `eless` do the "right thing" when it receives data from a pipe, because the piped data gets saved to a temp file which then gets opened by emacs behind the scenes. 

So emacs has no clue about the original command that generated that piped data.

You can read about the logic dealing with piped data here: https://eless.scripter.co/#output-is-going-to-stdout-input-is-coming-from-a-pipe

If you can get the eless script working for your usecase, please open your PR.

**Comment by kaushalmodi** [comment]
@jibal 

>This violates all sorts of software engineering principles, especially on a system based on composable components like UNIX. 

I developed this utility to serve a purpose i.e. to use eless aka Emacs for all the things where I used `cat` or `less`. It doesn't follow any principle.. Like any Emacs user, I am trying to do all things in Emacs 😄 

PR is welcome!

**Comment by jibal** [comment]
"to use eless aka Emacs for all the things where I used cat or less"

But it can't be used as such a replacement when the input is from man and any flags or a local file was passed to man ... which means that it can't be used generally as the value of $PAGER. This is why principles matter.

"I am trying to do all things in Emacs"

But it's not doing things _in_ Emacs ... that would mean using term or shell in an emacs window. Or maybe going halfway and using emacsclient so at least it's using one's current emacs session.

Anyway ... in most cases you just hand the temp file to emacs, but when it's a man page you invoke the emacs man command with the name of the manual page ... I think that's a mistake. I changed it to just hand emacs the temp file and that works for me. It doesn't use the Man-overstrike face but I'm fine with that. I may make some other tweaks to my copy (e.g., I plan to change it to be able to load a config file) ... it's just a single file shell script which I will drop into my own bin ... it's a good idea but I have serious issues with the implementation so I don't plan to submit a PR.
